extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java

```diff
@@ -229,6 +205,31 @@ BatchMediaItemResponse createPhoto(NewMediaItemUpload newMediaItemUpload) throws
     }
   }
 
+  private HttpResponse handleHttpResponseException(HttpRequest httpRequest, HttpResponseException e)
+      throws IOException {
+    // if the response is "unauthorized", refresh the token and try the request again
+    if (e.getStatusCode() == 401) {
+      monitor.info(() -> "Attempting to refresh authorization token");
+      // if the credential refresh failed, let the error bubble up via the IOException that gets
+      // thrown
+      credential = credentialFactory.refreshCredential(credential);
+      monitor.info(() -> "Refreshed authorization token successfuly");
+
+      // if the second attempt throws an error, then something else is wrong, and we bubble up the
+      // response errors
+      return httpRequest.execute();
```

Comment:
I think you need to recraft the request here (or update the access token that is in the request) because the credential has been refreshed.

Reply:
Good catch!